src/main/java/com/example/videotecha/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,49 @@
+package com.example.videotecha.service.impl;
+
+import com.example.videotecha.dto.UserDto;
+import com.example.videotecha.mapper.UserMapper;
+import com.example.videotecha.model.RegisteredUser;
+import com.example.videotecha.repository.RegisteredUserRepository;
+import com.example.videotecha.service.UserService;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class UserServiceImpl implements UserService {
+
+    private final RegisteredUserRepository registeredUserRepository;
```

Comment:
Generally not a bad idea to keep same name for all things related to 1 entity. Eg. UserService, UserRepository etc.